magit-todos.el

```diff
@@ -363,6 +363,12 @@ from the \"topic2\" branch, this option could be set to
   "Show submodule to-do list."
   :type 'boolean)
 
+(defcustom magit-todos-trim-filename-fn nil
+  "Apply function to file name to change its representation.
+Function takes relative file path as an argument and must return string."
+  :type '(choice (const :tag "Don't trim" nil)
+                 (function :tag "Function" f-filename)))
```

Comment:
Since the choice here is a specific function, this should be `(function-item :tag "Filename-only" f-filename)`.  Then a third choice should be `(function :tag "Custom function")`.  See Elisp manual section 15.4.2 Composite Types.